### PR TITLE
fix: (platform) Bug fix for multiple menu triggers with same menu

### PR DESCRIPTION
**Pull request by KevinOkamoto**
#### Please provide a link to the associated issue.
#2371 
#2539 

#### Please provide a brief summary of this pull request.

For a scenario where a single menu is referenced by multiple trigger via the `menuTriggerFor` directive, there is a bug when clicking from one button to the next.

Clicking on the first button will open the menu. Clicking on the second button will then close the menu, without opening it from the second trigger element. This happens because the `closeMenu` method invoked from an "outside" click is resolved after the open menu call from the second button click, so the menu does not open the second time.

The solution was to add a `setTimeout` delay on the open menu call to insure it happens after the close menu is resolved.

Because I added a `setTimeout` I needed to add `fakeAsync` to all unit tests which make use of a menu trigger.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

### CODE REVIEW

File: libs/platform/src/lib/components/menu/menu-trigger.directive.ts [hunk 1]
```diff
@@ -124,7 +124,15 @@ export class MenuTriggerDirective implements OnDestroy, AfterContentInit {
...
     toggleMenu() {
-        this._isMenuOpen ? this.destroyMenu() : this.openMenu();
+        /**
+         * Need to add delay here to ensure that any "closeMenu" operation which
+         * has been invoked within an "outsideClickSubscription" gets resolved
+         * before "openMenu" is called. This can happen when there are multiple
+         * triggers for the same menu.
+         */
+        setTimeout(() => {
+            this._isMenuOpen ? this.destroyMenu() : this.openMenu();
```
Comment (fkolar) [inline libs/platform/src/lib/components/menu/menu-trigger.directive.ts:134]: Can you use tick() instead ?

Comment (KevinOkamoto) [reply libs/platform/src/lib/components/menu/menu-trigger.directive.ts:134]: I believe `tick()` is a method from Angular's testing module:
https://angular.io/api/core/testing/tick

I don't think there exist a `tick` function in native Javascript, unless you know of some other instance.

**Comment by droshev** [comment]
@KevinOkamoto the styling is still broken.
Screen Shot 2020-06-26 at 6 18 56 PM

**Comment by KevinOkamoto** [comment]
Menu styles have been fixed.

File: libs/platform/src/lib/components/menu/menu-trigger.directive.ts (hunk 1 above)
Comment (fkolar) [reply libs/platform/src/lib/components/menu/menu-trigger.directive.ts:134]: you are right. I though I was commenting test.

**Comment by stefanoScalzo** [comment]
When menu is open, tabbing does not work

**Comment by stefanoScalzo** [comment]
The cursor should change when hovering over the image (not sure if that is in the scope of this pr though)

**Comment by KevinOkamoto** [comment]
@stefanoScalzo Tabbing issue has been fixed.

**Comment by InnaAtanasova** [comment]
There are some minor styling issues that I noticed:
1. When an item is selected from the dropdown, the buttons gets a solid blue line:
Screen Shot 2020-07-31 at 3 30 41 PM
Screen Shot 2020-07-31 at 3 30 49 PM

2. In Cascading menu the menu items have rounded border corners:
Screen Shot 2020-07-31 at 3 31 29 PM
Screen Shot 2020-07-31 at 3 32 55 PM

3. In the example `Menu Within Scrolling Container` the Menu button looks a bit cut apart from having the solid blue border:
Screen Shot 2020-07-31 at 3 33 58 PM

File: libs/platform/src/lib/components/button/button.component.ts [hunk 2]
```diff
@@ -66,6 +66,13 @@ export class ButtonComponent extends BaseComponent implements AfterViewInit {
     @Output()
     buttonClicked = new EventEmitter();
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/button/button.component.ts:67]: missing type

File: libs/platform/src/lib/components/menu/menu-item.component.ts [hunk 3]
```diff
@@ -31,6 +31,8 @@ export class MenuItemComponent implements OnDestroy, FocusableOption {
 
     // Add Fundamental-Styles class for menu item
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/menu/menu-item.component.ts:32]: These shouldn't be normal comments but typedocs

File: libs/platform/src/lib/components/menu/menu-item.component.ts [hunk 4]
```diff
@@ -31,6 +31,8 @@ export class MenuItemComponent implements OnDestroy, FocusableOption {
...
     @HostBinding('class.fd-menu__item') menuItemClass = true;
+    @HostBinding('class.fd-menu__link') menuLinkClass = true;
+
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/menu/menu-item.component.ts:35]: All hostbindings should have typedocs

File: libs/platform/src/lib/components/menu/menu-trigger.directive.ts [hunk 5]
```diff
@@ -54,7 +54,7 @@ export class MenuTriggerDirective implements OnDestroy, AfterContentInit {
         private _viewContainerRef: ViewContainerRef,
         @Optional() @Self() private _menuItem: MenuItemComponent,
         @Optional() private _parentMenu: MenuComponent
-    ) {}
+    ) { }
 
     ngAfterContentInit() {
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/menu/menu-trigger.directive.ts:59]: should return void

File: libs/platform/src/lib/components/menu/menu-trigger.directive.ts [hunk 6]
```diff
@@ -124,7 +124,15 @@ export class MenuTriggerDirective implements OnDestroy, AfterContentInit {
     }
 
     toggleMenu() {
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/menu/menu-trigger.directive.ts:126]: missing return type

File: libs/platform/src/lib/components/menu/menu-trigger.directive.ts [hunk 7]
```diff
@@ -124,7 +124,15 @@ export class MenuTriggerDirective implements OnDestroy, AfterContentInit {
...
+         * has been invoked within an "outsideClickSubscription" gets resolved
+         * before "openMenu" is called. This can happen when there are multiple
+         * triggers for the same menu.
+         */
+        setTimeout(() => {
+            this._isMenuOpen ? this.destroyMenu() : this.openMenu();
+        }, 0);
     }
 
     openMenu() {
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/menu/menu-trigger.directive.ts:138]: missing return type

File: libs/platform/src/lib/components/button/button.component.ts (hunk 2 above)
Comment (KevinOkamoto) [reply libs/platform/src/lib/components/button/button.component.ts]: Added missing type.

File: libs/platform/src/lib/components/menu/menu-item.component.ts (hunk 4 above)
Comment (KevinOkamoto) [reply libs/platform/src/lib/components/menu/menu-item.component.ts:38]: Added typedocs.

File: libs/platform/src/lib/components/menu/menu-trigger.directive.ts (hunk 5 above)
Comment (KevinOkamoto) [reply libs/platform/src/lib/components/menu/menu-trigger.directive.ts]: Added return type.

File: libs/platform/src/lib/components/menu/menu-trigger.directive.ts (hunk 6 above)
Comment (KevinOkamoto) [reply libs/platform/src/lib/components/menu/menu-trigger.directive.ts]: Added return type.

File: libs/platform/src/lib/components/menu/menu-item.component.ts (hunk 3 above)
Comment (KevinOkamoto) [reply libs/platform/src/lib/components/menu/menu-item.component.ts]: Added typedocs.

File: libs/platform/src/lib/components/menu/menu-trigger.directive.ts (hunk 7 above)
Comment (KevinOkamoto) [reply libs/platform/src/lib/components/menu/menu-trigger.directive.ts]: Added return type.

**Comment by KevinOkamoto** [comment]
@InnaAtanasova I was able to fix the "blue" outline on focus of the `ftp-button` component host element.

**Comment by InnaAtanasova** [comment]
Screen Shot 2020-08-04 at 4 18 25 PM

@KevinOkamoto What about this issue?

**Comment by KevinOkamoto** [comment]
@InnaAtanasova The round corner issue has been fixed.

**Pull request merged**